File: blog-sever/server-jpa/src/main/kotlin/com/site/blog/model/entity/SysDictData.kt
package com.site.blog.model.entity


import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.*
import org.hibernate.Hibernate
import java.time.LocalDateTime

/**
 * @author yanni
 * @date time 2022/6/17 12:33
 * @modified By:
 */
@Entity
data class SysDictData (

    var typeId: String? = null,
    var value: String? = null,
    var code: String? = null,
    var sort: Int? = null,
    var remark: String? = null,
    var status: Boolean? = null,

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    var createTime: LocalDateTime? = null,

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    var updateTime: LocalDateTime? = null,
) {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false)
    open var id: Long? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        other as SysDictData

        return id != null && id == other.id
    }

    override fun hashCode(): Int = javaClass.hashCode()

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(id = $id , typeId = $typeId , value = $value , code = $code , sort = $sort , remark = $remark , status = $status , createTime = $createTime , updateTime = $updateTime )"
    }
}

File: BlogApi/BlogApi/Models/SysDictData.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BlogApi.Models;

[Table("sys_dict_data")]
public class SysDictData
{
    [Key]
    [Column("id")]
    public string Id { get; set; } = Guid.NewGuid().ToString("N");

    [Column("type_id")]
    public string TypeId { get; set; } = string.Empty;

    [Column("value")]
    public string Value { get; set; } = string.Empty;

    [Column("code")]
    public string Code { get; set; } = string.Empty;

    [Column("sort")]
    public int Sort { get; set; }

    [Column("remark")]
    public string? Remark { get; set; }

    [Column("status")]
    public bool Status { get; set; }

    [Column("create_time")]
    public DateTime? CreateTime { get; set; }

    [Column("update_time")]
    public DateTime? UpdateTime { get; set; }
}
